main/svgparser.js

```diff
@@ -1572,6 +1508,85 @@ export class SvgParser {
 			if (command=='M' || command=='m') x0=x, y0=y;
 		}
 
+		// After generating the polygon points, check for self-intersection.
+		if (poly.length > 0) {
+			let uniquePolyForCheck = poly.slice();
+
+			// Remove duplicate last point if it's the same as the first (for closed paths)
+			if (uniquePolyForCheck.length > 1 && GeometryUtil.almostEqualPoints(uniquePolyForCheck[0], uniquePolyForCheck[uniquePolyForCheck.length - 1], this.conf.toleranceSvg)) {
+				uniquePolyForCheck.pop();
+			}
+
+      if (uniquePolyForCheck.length >= 3 && this.isPolygonSelfIntersecting(uniquePolyForCheck)) {
+				console.warn('SVG Import: Path was skipped due to self-intersection. Source path d:', path.getAttribute('d') || 'N/A');
+        console.trace();
+				return []; // Return an empty array to indicate this path should be skipped
+			}
+		}
+
 		return poly;
 	};
 }
+
+SvgParser.prototype.isPolygonSelfIntersecting = function(uniquePolygonPoints) {
```

Comment:
1. Can we make this a member of the Polygon class?
2. Can we add a unit test or two that demonstrates the correctness of the implementation? It looks reasonable to me, but there are some optimizations I can envision (I don't think we need to check for crossing of the lines formed from points (i, i+1) and (i+1, i+2) for example) and it'd be nice to have a test that shows whether those optimizations are valid.